Doras/src/app/roles/roles.component.ts

```diff
@@ -1,72 +1,86 @@
 import { Job } from '../models/job';
 import { ResponsibilitiesComponent } from '../responsibilities/responsibilities.component'
 import { MatDialog } from '@angular/material/dialog';
 import { SpecificationComponent } from '../specification/specification.component';
 import { TrainingPopupComponent } from '../training-popup/training-popup.component';
 import { Training } from '../models/training';
-
-
-import { SwitchBoardService } from '../switch-board.service';
 import { Competency } from '../models/Competency';
 import { BandCompetenciesComponent } from '../band-competencies/band-competencies.component';
+import { OtherJobsComponent } from '../other-jobs/other-jobs.component'
+import { SwitchBoardService } from '../switch-board.service';
+import { Subscription } from 'rxjs';
 
 @Component({
   selector: 'app-roles',
   templateUrl: './roles.component.html',
   styleUrls: ['./roles.component.css']
 })
 export class RolesComponent implements OnInit {
-  div: HTMLElement;
-  public bands: Band[];
-  public jobBandArray: any[];
   public pageCount = 0;
-  public firstJob: Job[];
-  public secondJob: Job[];
-  public thirdJob: Job[];
-  public firstTraining: Training[];
-  public secondTraining: Training[];
-  public thirdTraining: Training[];
+  public jobBandArray: any[];
+  public jobRole: any[];
+  public otherBands: any[];
   public firstCompetency: Competency[];
   public secondCompetency: Competency[];
   public thirdCompetency: Competency[];
+  public firstTraining: Training[];
+  public secondTraining: Training[];
+  public thirdTraining: Training[];
 
-
-  get capability(): Capability | null {
-    return this.dataTransferService.capability;
+  constructor(private data: DataService, private switchBoard: SwitchBoardService, public dialog: MatDialog, private config: NgbCarouselConfig) {
   }
 
-  constructor(private data: DataService, private dataTransferService: DataTransferService, public dialog: MatDialog) { }
-
   open(data: any[], component: any): void {
     this.dialog.open(component, {
       data: { data: data }
     });
   }
-
+  subJobRole: Subscription
+  subBand: Subscription
+  subOtherBand: Subscription
+  subFirstComp: Subscription
   ngOnInit() {
-    this.data.getBand().subscribe(c => {
-      this.bands = c.reverse();
-
-      const tempArray: any[] = [];
-      const step = 3;
-      let nextSplitValue: number;
-      const length: number = this.bands.length;
-      for (let i = 0; i < length; i += step) {
-        nextSplitValue = i + 3;
-        tempArray.push(this.bands.slice(i, nextSplitValue));
-      }
-      this.jobBandArray = tempArray;
+    this.subJobRole = this.switchBoard.job$.subscribe((c) => {
+      this.jobRole = c;
+    });
+
+    this.subBand = this.switchBoard.band$.subscribe((c) => {
+      this.jobBandArray = c;
     });
+
+    this.subOtherBand = this.switchBoard.otherBand$.subscribe((c) => {
+      this.otherBands = c;
+    });
+
+    this.data.getCompetenciesBand(9).subscribe(c => {
```

Comment:
what is going on here? 7,8,9? is there not a better way to do this? what are these values suppose to mean? 👎